viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
+        return avg_rating
 
-
+def get_most_watched_genre(user_data):
+    most_watched_genre = ""
+    list_of_genre = list()
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        for i in range(len(user_data["watched"])):
+            list_of_genre.append(user_data["watched"][i]["genre"])
+        most_watched_genre = statistics.mode(list_of_genre)
+        return most_watched_genre
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
-
+def get_unique_watched(user_data):
+    friends_movie_list = list()
+    unique_movies = list()
+    if len(user_data["watched"]) == 0:
+        unique_movies = []
+        return unique_movies
+    else:
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_movie_list.append(user_data["friends"][i]["watched"][j]["title"])
+        friends_movie_set = set(friends_movie_list)
+        for k in range(len(user_data["watched"])):
+            if user_data["watched"][k]["title"] not in friends_movie_set:
+                unique_movies.append(user_data["watched"][k])
+        return unique_movies
+    
+def get_friends_unique_watched(user_data):
+    friends_movie_list = list()
+    friends_unique_movies = list()
+    if len(user_data["watched"]) == 0:
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_movie_list.append(user_data["friends"][i]["watched"][j])
+        return friends_unique_movies
+    else:
+        for i in range(len(user_data["friends"])):
+            for j in range(len(user_data["friends"][i]["watched"])):
+                friends_movie_list.append(user_data["friends"][i]["watched"][j])
+        for k in range(len(friends_movie_list)):
+            if (friends_movie_list[k] not in user_data["watched"]) and ((friends_movie_list[k] not in friends_unique_movies)):
+                friends_unique_movies.append(friends_movie_list[k])
+        return friends_unique_movies#_set
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
-
+def get_available_recs(user_data):
+    list_of_recommended_movies = list()
+    unique_friends_movie = get_friends_unique_watched(user_data)
+    if not user_data["watched"] or not unique_friends_movie or not user_data["subscriptions"]:
+        return list_of_recommended_movies
+    else:
+        for i in range(len(unique_friends_movie)):
+            if unique_friends_movie[i]["host"] in user_data["subscriptions"]:
+                list_of_recommended_movies.append(unique_friends_movie[i])
+        if not list_of_recommended_movies:
+            return list_of_recommended_movies
+        else:
+            return list_of_recommended_movies

```

Comment:
Because the `if` and `else` contain the same code, we could remove those and just keep the line `return list_of_recommended_movies`